test_fms/diag_manager/test_diag_object_container.F90

```diff
@@ -0,0 +1,234 @@
+  type (fms_diag_object), pointer::   pobj                        !< A pointer to an object
+  integer, parameter :: num_objs = 10                             !< Total number of objects tested
+  integer ::  full_id_sum                                         !< Sum of all the possible object id values
+  integer :: sum                                                  !< Temp sum of vaalues of id sets
+  !!
+  integer :: ic_status                                            !< A status flag returned from container functions
+  integer :: ierr                                                 !< An error flag
+  !!
+  logical :: test_passed                                          !> Flag indicating if the test_passed
+  !! These fields below used to initialize diag object data. TBD
+  integer :: id
+  integer, dimension(2) :: axes
+  TYPE(time_type)  :: init_time
+  !!type (diag_fields_type)  :: diag_field
+  character(:), allocatable :: mname, vname, mname_pre, vname_pre
+  !!
+
+
+  test_passed = .true.  !! will be set to false if there are any issues.
+
+  call mpp_init(mpp_init_test_requests_allocated)
+  call mpp_io_init()
+  call mpp_set_stack_size(3145746)
+
+  !! Ids will initally be from1 to num_objs, so :
+  full_id_sum = (num_objs * (num_objs + 1)) / 2
+
+  !!Create the container
+  container = FmsDiagObjectContainer_t()
+  !!In diag_manager, one module level container may be used instead of a local one like above.
+
+
+  !! Allocate some test objects.
+  !! NOTE: normally objects will be allocated one at a time with a stament like:
+  !!   allocate(pobj, source = fms_diag_object(argument list ))
+  !! or via constructor like :
+  !!   pobj => fms_diag_object(argument list )
+  !! Once the object ID is set, it should be inserted into the container and then the
+  !!   container will be considered the manager of that object and its memory (unless the object is removed).
+  !! Since type fms_diag_obj doesn't have a proper constructor yet, well be lazy by making array of objects
+  !! ( normal fixed size array the thing whose use we are replacing to begin with ) and consider these particular
+  !! objects to not be managed by the container.
+  allocate(obj_vec(num_objs))
+
+  !! Initialize each object and isnert into container one at a time.
+
+  if( container%size() /= 0) then
+    test_passed = .false.
+    call mpp_error(WARNING, "Container incorrect size. Expected 0 at start")
+  endif
+  mname_pre = "ATM"
+  vname_pre = "xvar"
+  do id = 1, num_objs
+    call combine_str_int(mname_pre, id, mname)
+    call combine_str_int(vname_pre, id, vname )
+
+    pobj => obj_vec( id ) !!Note use of pointer to obj.
+    call pobj%setID(id)
+
+    call pobj%register ("test_mod", vname, axes, init_time, "a_long_name")
+
+    !!Insert object into the container.
+    ic_status = container%insert(pobj%get_id(), pobj)
+    if(ic_status .ne. 0)then
+      test_passed = .false.
+      call mpp_error(FATAL, "Container Isertion ERROR for")
```

Comment:
It is there because :
 a) that is how the user shoud use the insert function, To that end I improved the documentaion
of function insert_diag_object.
 b)  Its a partial check of the stated function interface. 
I also improved grammar and spelling in the message.

Reply:
"Container Isertion ERROR for" should have an "n" ?
"Container I**n**sertion ERROR for"
Sorry, I should have been more explicit in my comment